misc/install_initrd.sh

```diff
@@ -0,0 +1,221 @@
+		myerror "Error: The ramdisk size must be a positive integer. Exiting..."
+	fi
+fi
+
+
+# Check for CentOS
+if hostnamectl | grep "CentOS Linux" >/dev/null 2>/dev/null ; then
+	echo " - CentOS detected!"
+
+	if [ -z $uninstall ] ; then
+
+		echo " - Installing by copying 96rapiddisk into /usr/lib/dracut/modules.d/..."
+		rm -rf /usr/lib/dracut/modules.d/96rapiddisk
+		cp -rf 96rapiddisk /usr/lib/dracut/modules.d/
+		echo " - Editing /usr/lib/dracut/modules.d/96rapiddisk/run_rapiddisk.sh..."
+		sed -i 's/RAMDISKSIZE/'${ramdisk_size}'/' /usr/lib/dracut/modules.d/96rapiddisk/run_rapiddisk.sh
+		sed -i 's,BOOTDEVICE,'${boot_device}',' /usr/lib/dracut/modules.d/96rapiddisk/run_rapiddisk.sh
+		echo " - chmod +x /usr/lib/dracut/modules.d/96rapiddisk/* ..."
+		chmod +x /usr/lib/dracut/modules.d/96rapiddisk/*
+	else
+		echo " - Uninstalling by removing dir /usr/lib/dracut/modules.d/96rapiddisk..."
+		rm -rf /usr/lib/dracut/modules.d/96rapiddisk
+	fi
+	
+	echo " - Running 'dracut --kver $kernel_version -f'"
+	#dracut --add-drivers "rapiddisk rapiddisk-cache" --kver "$kernel_version" -f
+	dracut --kver "$kernel_version" -f
+	
+	echo " - Done under CentOS."
+	exit 0
+fi
+
+# Ubuntu
+
+if [ -d /etc/initramfs-tools/hooks ] && [ -d /etc/initramfs-tools/scripts/init-premount ] ; then
+	hook_dest=/etc/initramfs-tools/hooks/rapiddisk_hook
+	bootscript_dest=/etc/initramfs-tools/scripts/init-premount/rapiddisk
+elif [ -d /usr/share/initramfs-tools/hooks ] && [ -d /usr/share/initramfs-tools/scripts/init-premount ] ; then
+	hook_dest=/usr/share/initramfs-tools/hooks/rapiddisk_hook
+	bootscript_dest=/usr/share/initramfs-tools/scripts/init-premount/rapiddisk
+else
+	myerror "Error: I can't find any suitable place for initramfs scripts."
+fi
+
+
+if [ ! -z $uninstall ] ; then
+	echo " - Starting uninstall process..."
+	if [ -x "$hook_dest" ] || [ -x "$bootscript_dest" ] ; then
+		echo " - Uninstalling scripts..."
+		rm -f "$hook_dest" 2>/dev/null
+		rm -f "$bootscript_dest" 2>/dev/null
+	fi
+	#if lsinitramfs /boot/initrd.img-"$kernel_version" | grep rapiddisk >/dev/null 2>/dev/null ; then
+	#	echo ""
+	#	echo "Error: /boot/initrd.img-$kernel_version still contains rapiddisk stuff."
+	#	echo
+	#	exit 1
+	#fi
+	finalstuff
+	exit 0	
+
+fi
+
+# TODO this check must be improved
+if ! echo "$boot_device" | grep -E '^/dev/[[:alpha:]]{1,4}[[:digit:]]{0,99}$' >/dev/null 2>/dev/null ; then
+        myerror "Error: boot_device must be in the form '/dev/xxx' or '/dev/xxxn with n as a positive integer. Exiting..."
+fi
+
+if [ ! -x "${cwd}/hooks/rapiddisk_hook" ] || [ ! -x "${cwd}/scripts/init-premount/rapiddisk" ] ; then
+	myerror "Error: I can't find the scripts to be installed. Exiting..."
+fi
+
+##############################################################################
+# TODO The following piece of code was there to check if the user specified
+# the same device present in /etc/fstab - but if in fstab UUIDs are used, or
+# the LABEL= syntax, it looses meaning.
```

Comment:
minor minor minor detail: misspelling: supposed to be "loses"

Reply:
I fixed it!